lib/service/mock_api_service.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:math';
+
+import 'package:dartz/dartz.dart';
+import 'package:injectable/injectable.dart';
+
+@LazySingleton()
+class MockApiService {
+
+  List<String> get items => [for (int i = 0; i < 9999999; i++) 'My List Item nr $i'];
```

Comment:
We should investigate turning this into a generator function to improve performance by not generating values except if needed. this might also be a good point to precompute the index so that it's available when needed
```suggestion
  Iterable<String> get items => [for (int i = 0; i < 9999999; i++) 'My List Item nr $i'];
```

Reply:
Another slight improvement option would be to switch to a List.generate constructor as that is slightly more optimize as it can currently precompute the List size ahead of time